simple_history/management/commands/clean_duplicate_history.py

```diff
@@ -0,0 +1,118 @@
+from django.utils import timezone
+from django.db import transaction
+
+from . import populate_history
+from ... import models, utils
+from ...exceptions import NotHistoricalModelError
+
+get_model = populate_history.get_model
+
+
+class Command(populate_history.Command):
+    args = "<app.model app.model ...>"
+    help = (
+        "Scans HistoricalRecords for identical sequencial entries "
+        "(duplicates) in a model and deletes them."
+    )
+
+    DONE_CLEANING_FOR_MODEL = "Removed {count} historical records for {model}\n"
+
+    def add_arguments(self, parser):
+        parser.add_argument("models", nargs="*", type=str)
+        parser.add_argument(
+            "--auto",
+            action="store_true",
+            dest="auto",
+            default=False,
+            help="Automatically search for models with the HistoricalRecords field "
+            "type",
+        )
+        parser.add_argument(
+            "-d", "--dry", action="store_true", help="Dry (test) run only, no changes"
+        )
+        parser.add_argument(
+            "-m", "--minutes", type=int, help="Only search the last MINUTES of history"
+        )
+
+    def handle(self, *args, **options):
+        self.verbosity = options["verbosity"]
+
+        to_process = set()
+        model_strings = options.get("models", []) or args
+
+        if model_strings:
+            for model_pair in self._handle_model_list(*model_strings):
+                to_process.add(model_pair)
+
+        elif options["auto"]:
+            for model in models.registered_models.values():
+                try:  # avoid issues with mutli-table inheritance
+                    history_model = utils.get_history_model_for_model(model)
+                except NotHistoricalModelError:
+                    continue
+                to_process.add((model, history_model))
+            if not to_process:
+                if self.verbosity >= 1:
+                    self.stdout.write(self.NO_REGISTERED_MODELS)
+
+        else:
+            if self.verbosity >= 1:
+                self.stdout.write(self.COMMAND_HINT)
+
+        self._process(to_process, date_back=options["minutes"], dry_run=options["dry"])
+
+    def _process(self, to_process, date_back=None, dry_run=True):
+        if date_back:
+            stop_date = timezone.now() - timezone.timedelta(minutes=date_back)
```

Comment:
Should timezone.now() be specified on initialization rather than at this point? Maybe move it to the top of `handle`?

Reply:
that point is actually called only once per execution but I've moved it to the top of `handle`

Reply:
just undid this change as linting complains about `handle` becoming too complex. I think it might have been that that drove me to move it to `_process` in first place, can't remember...